### PR TITLE
Add MDACC Heatmap to study view

### PR DESCRIPTION
**Pull request by zhx828**
Reimplement https://github.com/cBioPortal/cbioportal/pull/1709/files following https://github.com/cBioPortal/cbioportal-frontend/pull/252

This solves issue https://github.com/cBioPortal/cbioportal/issues/4833

### CODE REVIEW

**Comment by zhx828** [comment]
@kalletlak @alisman could you take a look why `mocha` complains about `Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.`?

**Comment by zhx828** [comment]
@ChrisWakefield this is the pr to new study view

**Comment by kalletlak** [comment]
@zhx828 from the test logs 
`LOG: 'Warning: calling stubbed method request.get, which returns a promise that never resolves.'`

File: src/shared/lib/MDACCUtils.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+export async function getHeatmapMeta(url: string): Promise<string[]> {
+    let result: string[] = [];
+    try {
+        let resp: any = await request.get(url);
+
+        const parsedResp: any = JSON.parse(resp.text);
+
+        // filecontent array is serialized :(
+        result = JSON.parse(parsedResp.fileContent);
```
Comment (kalletlak) [inline src/shared/lib/MDACCUtils.ts]: may be 
```
    return JSON.parse(parsedResp.fileContent);
} catch (e) {
    return []
}
```

**Comment by zhx828** [comment]
- [ ] The loading gif in latest rc branch will not disappear in the MDACC tab

File: src/pages/patientView/clinicalInformation/PatientViewPageStore.ts [hunk 2]
```diff
@@ -304,16 +305,8 @@ export class PatientViewPageStore {
...
         invoke: async() => {
-
-            let resp: any = await request.get(`//bioinformatics.mdanderson.org/dyce?app=chmdb&command=participant2maps&participant=${this.patientId}`);
-
-            const parsedResp: any = JSON.parse(resp.text);
-
-            // filecontent array is serialized :(
-            const fileContent: string[] = JSON.parse(parsedResp.fileContent);
-
+            const fileContent: string[] = await getHeatmapMeta(`//bioinformatics.mdanderson.org/participant2maps?participant=${this.patientId}`);
```
Comment (alisman) [inline src/pages/patientView/clinicalInformation/PatientViewPageStore.ts]: can that service support both https and http?

File: src/shared/lib/MDACCUtils.ts (hunk 1 above)
Comment (alisman) [reply src/shared/lib/MDACCUtils.ts]: agree

File: src/pages/patientView/clinicalInformation/PatientViewPageStore.ts (hunk 2 above)
Comment (zhx828) [reply src/pages/patientView/clinicalInformation/PatientViewPageStore.ts]: Yes

File: src/shared/lib/MDACCUtils.ts (hunk 1 above)
Comment (zhx828) [reply src/shared/lib/MDACCUtils.ts]: Done

**Pull request merged**